text/0000-freeze-uninit.md

```diff
@@ -0,0 +1,208 @@
+- Feature Name: `freeze_bytes`
+- Start Date: 2024-02-13
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- Feature Name: `freeze_uninit`
+
+# Summary
+[summary]: #summary
+
+Defines a language primitive for freezing uninitialized bytes, and library functions that use this language primitive.
+
+# Motivation
+[motivation]: #motivation
+
+In Rust, it is generally undefined behaviour to read uninitialized memory. Most types disallow uninitialized bytes as part of their validity invariant. 
+While in most cases, this is not a problem, as a temporarily held (or externally checked) uninitialized value can be stored as `MaybeUninit<T>`, in some rare cases it can be useful or desireable to handle uninitialized data as a "Don't Care" value, while still doing typed operations, such as arithmetic. 
+Freeze allows these limited Rust Programs to convert uninitialized data into useless-but-initialized bytes.
+
+Examples of uses:
+1. The major use for freeze is to read padding bytes of structs. This can be used for [zero-copy serialization](https://docs.rs/abomonation/latest/abomonation/) or a [generic wrapper arround standard atomic types](https://docs.rs/atomic/latest/atomic/struct.Atomic.html). 
```

Comment:
```suggestion
1. The major use for freeze is to read padding bytes of structs. This can be used for [zero-copy serialization](https://docs.rs/abomonation/latest/abomonation/) or a [generic wrapper around standard atomic types](https://docs.rs/atomic/latest/atomic/struct.Atomic.html). 
```